### PR TITLE
Fix #292 - Affiliate & Category (drilldown) views don't work

### LINKED ISSUE #292: SI: Viewset Affiliate & Category (drilldown) views don't work
## Describe the bug
When any of the following views are used with the Mandeeps Porto5 viewset, they don't work and instead throw JS exceptions.

## Software Versions
- DNN:  09.06.01
- Hotcakes:  03.04.00

## To Reproduce
1. Install DNN 
2. Install the Porto 5.x theme
3. Install Hotcakes
4. Add sample products and categories
5. Enable the affiliate program.
6. Add a new page that's visible to everyone.
7. Add the Affiliate Registration module to this new page.
8. Add a new page that's visible to at least Registered Users.
9. Add the Affiliate Dashboard module to this new page.  
10. In a new browser or incognito window, attempt to register as a new affiliate, filling in all fields except for the Referral Affiliate ID.  

Steps to Reproduce the Behavior:
1. Follow steps 1 through 4 above.  
2. In any instance of the category viewer module, configure it to use the drilldown view in module settings.

## Expected behavior
The drilldown and affiliate views work without reporting JS exceptions.  

## Actual behavior
There are JS exceptions thrown in the developer console.

### PR DESCRIPTION
**Pull request by erw13n**
Fix #292 include :
- Javascript issue due missing validation for Confirm Terms
- Term Popup show nothing event if affiliate agreement terms been fill in

Tested on :
- Default Theme
- Porto Theme

### CODE REVIEW

**Comment by hismightiness** [comment]
Please merge your viewset changes into the developer examples and other viewsets.   

https://github.com/HotcakesCommerce/hotcakes-commerce-core/tree/development/DevSamples/MyViewSet/Portals/_default/HotcakesViews  
https://github.com/HotcakesCommerce/hotcakes-commerce-core/tree/development/Website/Portals/_default/HotcakesViews

File: Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs [hunk 1]
```diff
@@ -166,9 +166,12 @@ public string ToJson()
         [RegularExpression(@"[-\w]*", ErrorMessage = "Only alpha-numeric and '-', '_' characters are allowed")]
         public string ReferralAffiliateId { get; set; }
 
+        public bool IsTrue => true;
```
Comment (hismightiness) [inline Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs:169]: Please change this to a standard property.  The build process won't recognize this version of C# (right now).

**Comment by erw13n** [comment]
OK, I'll update the other viewset.

File: Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs (hunk 1 above)
Comment (WillStrohl) [reply Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs]: @erw13n I don't see your update on this one yet?

**Comment by WillStrohl** [comment]
Please see the other code review note as well.  This PR is blocked right now.  :(

File: Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs (hunk 1 above)
Comment (erw13n) [reply Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs]: Understood, I'll update it.

File: Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs [hunk 2]
```diff
@@ -166,7 +166,7 @@ public string ToJson()
         [RegularExpression(@"[-\w]*", ErrorMessage = "Only alpha-numeric and '-', '_' characters are allowed")]
         public string ReferralAffiliateId { get; set; }
 
-        public bool IsTrue => true;
+        public bool IsTrue { get { return true; } }
```
Comment (erw13n) [inline Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs:169]: @WillStrohl It's been update in here. 
Include in other commit.

File: Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs (hunk 1 above)
Comment (erw13n) [reply Website/DesktopModules/Hotcakes/Core/Models/AffiliateViewModel.cs]: I already update it, its on commit of "Update the other Viewset" below.

**Review by erw13n — COMMENTED** [review]
@WillStrohl This commit contain change for other viewset: MyBootstrap3, MyLegacy, MyViewset, Porto5 & SocialSpokes

**Pull request merged**